liquibase-couchbase/src/test/java/liquibase/ext/couchbase/operator/ChangeLogOperatorTest.java

```diff
@@ -0,0 +1,216 @@
+import liquibase.ext.couchbase.changelog.Context;
+import liquibase.ext.couchbase.changelog.CouchbaseChangeLog;
+import liquibase.ext.couchbase.database.CouchbaseConnection;
+import liquibase.ext.couchbase.database.CouchbaseLiquibaseDatabase;
+import liquibase.ext.couchbase.provider.ContextServiceProvider;
+import liquibase.ext.couchbase.provider.ServiceProvider;
+import lombok.NonNull;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+import static common.constants.TestConstants.TEST_BUCKET;
+import static common.constants.TestConstants.TEST_SCOPE;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptySet;
+import static java.util.Collections.singletonList;
+import static liquibase.ext.couchbase.operator.ChangeLogOperator.SELECT_ALL_CHANGELOGS_N1QL;
+import static liquibase.ext.couchbase.operator.ChangeLogOperator.SELECT_LAST_CHANGE_SET_ID_N1QL;
+import static liquibase.ext.couchbase.operator.ChangeLogOperator.SELECT_LAST_ORDER_N1QL;
+import static liquibase.ext.couchbase.provider.ServiceProvider.CHANGE_LOG_COLLECTION;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class ChangeLogOperatorTest {
+
+    @Mock
+    private ServiceProvider serviceProvider;
+
+    @Mock
+    private CouchbaseLiquibaseDatabase database;
+
+    @Mock
+    private CouchbaseConnection connection;
+
+    @Mock
+    private Cluster cluster;
+
+    @Mock
+    private QueryIndexManager queryIndexManager;
+
+    @Mock
+    private Scope testScope;
+
+    @Mock
+    private Collection collection;
+
+    @Mock
+    private QueryResult queryResult;
+
+    @Mock
+    private ChangeSet changeSet;
+
+    @InjectMocks
+    private ChangeLogOperator changeLogOperator;
+
+    @Test
+    void Should_create_changelog_collection() {
+        when(serviceProvider.getScopeOfCollection(CHANGE_LOG_COLLECTION)).thenReturn(testScope);
+        when(database.getConnection()).thenReturn(connection);
+        when(connection.getCluster()).thenReturn(cluster);
+        when(cluster.queryIndexes()).thenReturn(queryIndexManager);
+        when(testScope.name()).thenReturn(TEST_SCOPE);
+        when(testScope.bucketName()).thenReturn(TEST_BUCKET);
+
+        changeLogOperator.createChangeLogCollection();
+
+        verify(queryIndexManager, times(1)).createPrimaryIndex(eq(TEST_BUCKET), any(CreatePrimaryQueryIndexOptions.class));
```

Comment:
`times(1)` is extra as it default behaviour for `verify(call)`

Reply:
Thank you. Done